src/llamafactory/train/trainer_utils.py

```diff
@@ -425,3 +425,61 @@ def get_batch_logps(
     labels[labels == label_pad_token_id] = 0  # dummy token
     per_token_logps = torch.gather(logits.log_softmax(-1), dim=2, index=labels.unsqueeze(2)).squeeze(2)
     return (per_token_logps * loss_mask).sum(-1), loss_mask.sum(-1)
+
+
+def glm4v_compute_loss_warpper(old_compute_loss):
```

Comment:
Would you please let me know if this is necessary? 
Did you try to use the forward function in the modeling_chatglm of glm_4v?

Reply:
Overloading the `compute_loss` function is intended to address the padding issue. Specifically, this is due to the peculiar behavior of the glm4v model. Firstly, any image in the glm4v model is processed as 1600 embedded tokens, but in the official `forward` method of the glm4v model, the following occurs: after `input_ids` are transformed into embedded tokens, they are automatically padded. However, the attention mask used for loss calculation and attention computation, as well as the label masks/labels, are not automatically padded by the glm4v model. Handling this at the data processing stage might require overloading a larger number of functions and adding parameters designed specifically for glm4v. To expedite the process, I adopted this peculiar overloading method. At which stage do you think we should handle this issue?

Reply:
Seems glm4v dev team had notice this issue.Here is the related issus and code https://github.com/THUDM/GLM-4/issues/264 [huggingface code](https://huggingface.co/THUDM/glm-4v-9b/blob/1127073f37ed50ae3be6b839aaad828d44bc920d/modeling_chatglm.py#L1081)

Reply:
They updated code two days ago.I think we can test can remove this.

Reply:
Okay, thanks for noticing that!!🤗